src/main/java/org/mockito/internal/session/DefaultMockitoSessionBuilder.java

```diff
@@ -0,0 +1,33 @@
+package org.mockito.internal.session;
+
+import org.mockito.MockitoSession;
+import org.mockito.internal.framework.DefaultMockitoSession;
+import org.mockito.internal.util.ConsoleMockitoLogger;
+import org.mockito.quality.Strictness;
+import org.mockito.session.MockitoSessionBuilder;
+
+public class DefaultMockitoSessionBuilder implements MockitoSessionBuilder {
+
+    private Object testClassInstance;
+    private Strictness strictness;
+
+    @Override
+    public MockitoSessionBuilder initMocks(Object testClassInstance) {
+        this.testClassInstance = testClassInstance;
+        return this;
+    }
+
+    @Override
+    public MockitoSessionBuilder strictness(Strictness strictness) {
+        this.strictness = strictness;
+        return this;
+    }
+
+    @Override
+    public MockitoSession startMocking() {
+        //Configure default values
+        Object effectiveTest = this.testClassInstance == null ? new Object() : this.testClassInstance;
```

Comment:
>I would rather move this argument to the constructor

To the constructor of DefaultMockitoSessionBuilder? Are you advocating for API modification:

```java
//current
Mockito.session().initMocks(this).strictness(Strictness.WARN).startMocking();

//suggested by Tim
Mockito.session().strictness(Strictness.WARN).startMocking(this);
//or
Mockito.session(this).strictness(Strictness.WARN).startMocking();
```

>Moreover, it does not notify framework authors when they are using the listener

This feedback is unclear to me.